core/DataAccess/Model.php

```diff
@@ -984,4 +1018,14 @@ private function getHashFromDoneFlag($doneFlag)
         preg_match('/^done([a-zA-Z0-9]+)/', $doneFlag, $matches);
         return $matches[1] ?? '';
     }
+
+    private function doRequestedRecordsLookNumeric(array $requestedRecords): bool
+    {
+        foreach ($requestedRecords as $record) {
+            if (preg_match('/^nb_/', $record)) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Is the logic here that a single `nb_` in a list of 10 metrics is better than the rest? 
Should we count `nb_` and if it's more than half then return true, otherwise false? Would it have any impact?

Reply:
> Is the logic here that a single nb_ in a list of 10 metrics is better than the rest?

The logic here is: is there a record that is likely a numeric? If so look in the archive_numeric table first. Otherwise archive_blob. If we find all the data in one table, we don't have to query the other.

Note: this was requested by thomas above.

> Should we count nb_ and if it's more than half then return true, otherwise false? Would it have any impact?

No.

Reply:
I get the mechanism, I was more thinking whether it's the correct behaviour that a single 'likely' numeric record trumps all the rest regardless of the possibly way larger count of the non-numeric ones.

Reply:
If there's at least one numeric, then we have to look in the archive_numeric tables no matter what to check for it. The number of numeric records in the list doesn't change that.